native/common/jp_gc.cpp

```diff
@@ -0,0 +1,182 @@
+#include <Python.h>
+#include "jpype.h"
+#include "pyjp.h"
+#include "jp_reference_queue.h"
+
+#ifdef WIN32
+#define USE_PROCESS_INFO
+#include <Windows.h>
+#include <psapi.h>
+#elif __APPLE__
+#define USE_TASK_INFO
+#include <unistd.h>
+#include <sys/resource.h>
+#include <mach/mach.h>
+#else
+// Linux doesn't have an available rss tally so use mallinfo
+#define USE_MALLINFO
+#include <malloc.h>
+#endif
+#define DELTA_LIMIT 10*1024*1024l
+#define HARD_LIMIT 200*1024*1024l
+
+namespace
+{
+}
+
+size_t getWorkingSize()
+{
+	size_t current = 0;
+#if defined(USE_PROCESS_INFO)
+	PROCESS_MEMORY_COUNTERS pmc;
+	GetProcessMemoryInfo(GetCurrentProcess(), &pmc, sizeof (pmc));
+	current = (size_t) pmc.WorkingSetSize;
+
+#elif defined(USE_TASK_INFO)
+	struct mach_task_basic_info info;
+	mach_msg_type_number_t count = MACH_TASK_BASIC_INFO_COUNT;
+	if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO, (task_info_t) & info, &count) == KERN_SUCCESS)
+		current = (size_t) info.resident_size;
+	return (size_t) 0; /* query failed */
```

Comment:
you assign current, but then return 0 unconditionally?!

Reply:
Gack!